### ISSUE TITLE
When training the model, add the option "--eval", error

### ISSUE THREAD

**Issue by cena001plus** [PaddlePaddle/PaddleDetection#5128]
python -m paddle.distributed.launch --gpus 0,1,2,3 tools/train.py -c configs/picodet/picodet_s_416_voc_m.yml --eval
When training the model, add the option "--eval", an error occurs, after the model is saved, and the verification starts, the following problem will be prompted. I analyzed and found that gt_boxes is caused by a null value, but I don't know how this happened.

```
[01/18 17:34:37] ppdet.utils.checkpoint INFO: Save checkpoint: output/picodet_s_416_voc_m
yes, gt_boxes length is: 2
gt_boxes: Tensor(shape=[2, 0, 4], dtype=float32, place=CUDAPinnedPlace, stop_gradient=True,
       [[],

        []])


Traceback (most recent call last):
  File "tools/train.py", line 171, in <module>
    main()
  File "tools/train.py", line 167, in main
    run(FLAGS, cfg)
  File "tools/train.py", line 127, in run
    trainer.train(FLAGS.eval)

    return to_string(self)
  File "/home/anaconda3/envs/paddledet/lib/python3.7/site-packages/paddle/tensor/to_string.py", line 232, in to_string
    np_var = var.numpy()
RuntimeError: (PreconditionNotMet) Tensor's dimension is out of bound.Tensor's dimension must be equal or less than the size of its memory.But received  Tensor's dimension is d16 memory's size is 0.
  [Hint: Expected numel() * SizeOfType(type()) <= memory_size(), but received numel() * SizeOfType(type()):16 > memory_size():0.] (at /paddle/paddle/fluid/framework/tensor.cc:39)
```

**Comment by yghstill** [comment]
@cena001plus The model is not trained well, and there is no result output. If the data set is small, increase [snapshot_epoch](https://github.com/PaddlePaddle/PaddleDetection/blob/release/2.3/configs/picodet/picodet_s_416_coco.yml#L14), or load the pre-trained model on coco: 
```
-o pretrain_weights=https://paddledet.bj.bcebos.com/models/picodet_s_416_coco.pdparams
```

**Comment by cena001plus** [comment]
> @cena001plus The model is not trained well, and there is no result output. If the data set is small, increase [snapshot_epoch](https://github.com/PaddlePaddle/PaddleDetection/blob/release/2.3/configs/picodet/picodet_s_416_coco.yml#L14), or load the pre-trained model on coco:
> 
> ```
> -o pretrain_weights=https://paddledet.bj.bcebos.com/models/picodet_s_416_coco.pdparams
> ```

The complete data set is used for training, 500,000 data, but the problem still exists, do I still need to add the following commands? 
`-o pretrain_weights=https://paddledet.bj.bcebos.com/models/picodet_s_416_coco.pdparams`


still wrong when add this commdand,
`trainer.train(FLAGS.eval)
  File "/home/work/detection/PaddleDetection/ppdet/engine/trainer.py", line 450, in train
    self._eval_with_loader(self._eval_loader)
  File "/home/work/detection/PaddleDetection/ppdet/engine/trainer.py", line 474, in _eval_with_loader
    metric.update(data, outs)
  File "/home/work/detection/PaddleDetection/ppdet/metrics/metrics.py", line 270, in update
    print(gt_boxes[i])
  File "/home/anaconda3/envs/paddledet_m/lib/python3.7/site-packages/paddle/fluid/dygraph/varbase_patch_methods.py", line 482, in __str__
    return to_string(self)
  File "/home/anaconda3/envs/paddledet/lib/python3.7/site-packages/paddle/tensor/to_string.py", line 232, in to_string
    np_var = var.numpy()
RuntimeError: (PreconditionNotMet) Tensor's dimension is out of bound.Tensor's dimension must be equal or less than the size of its memory.But received  Tensor's dimension is d16 memory's size is 0.
  [Hint: Expected numel() * SizeOfType(type()) <= memory_size(), but received numel() * SizeOfType(type()):16 > memory_size():0.] (at /paddle/paddle/fluid/framework/tensor.cc:39)

INFO 2022-01-20 09:13:20,010 launch_utils.py:319] terminate process group gid:3368276
INFO 2022-01-20 09:13:20,012 launch_utils.py:319] terminate process group gid:3368281
INFO 2022-01-20 09:13:20,013 launch_utils.py:319] terminate process group gid:3368286
INFO 2022-01-20 09:13:24,018 launch_utils.py:340] terminate all the procs
ERROR 2022-01-20 09:13:24,019 launch_utils.py:603] ABORT!!! Out of all 4 trainers, the trainer process with rank=[0] was aborted. Please check its log.
INFO 2022-01-20 09:13:28,023 launch_utils.py:340] terminate all the procs
INFO 2022-01-20 09:13:28,023 launch.py:304] Local processes completed.
`

**Comment by cena001plus** [comment]
Should I modify your code? But I don't know if the problem arises?
It often leads to other problems after modification. In short, it cannot be solved. when len(gt_bboxes) not 0, but gt_boxes[i] is null?:


/PaddleDetection/ppdet/metrics/metrics.py"

**`         if len(gt_boxes)==0:  #add
            print("no")
        else:
            print("yes, gt_boxes length is:", len(gt_boxes))
            print("gt_boxes:",gt_boxes)
            for i in range(len(gt_boxes)):
                print(i)
                print(gt_boxes[i])`**[](url)

**Comment by cena001plus** [comment]
Should I modify your code? But I don't know if the problem arises?
It often leads to other problems after modification. In short, it cannot be solved. The code I modified is as follows:
/PaddleDetection/ppdet/metrics/metrics.py"

`  if len(gt_boxes)==0:  #add
            print("no")
        else:
            print("yes, gt_boxes length is:", len(gt_boxes))
            print("gt_boxes:",gt_boxes)
            for i in range(len(gt_boxes)):
                print(i)
                print(gt_boxes[i])
                
                gt_box = gt_boxes[i].numpy()
                h, w = scale_factor[i]
                gt_box = gt_box / np.array([w, h, w, h])`

**Comment by cena001plus** [comment]
Should I modify your code? But I don't know if the problem arises?
It often leads to other problems after modification. In short, it cannot be solved. The code I modified is as follows:
/PaddleDetection/ppdet/metrics/metrics.py"

`  if len(gt_boxes)==0:  #add
            print("no")
        else:
            print("yes, gt_boxes length is:", len(gt_boxes))
            print("gt_boxes:",gt_boxes)
            for i in range(len(gt_boxes)):
                print(i)
                print(gt_boxes[i])
                
                gt_box = gt_boxes[i].numpy()
                h, w = scale_factor[i]
                gt_box = gt_box / np.array([w, h, w, h])`

**Comment by cena001plus** [comment]
solved!
#5137

**Comment by 3244we** [comment]
> solved! #5137

How did you solve this problem？
